src/main/java/com/cloudogu/smeagol/SmeagolResource.java

```diff
@@ -24,19 +24,66 @@
 
 package com.cloudogu.smeagol;
 
+import de.otto.edison.hal.Embedded;
+import de.otto.edison.hal.HalRepresentation;
 import sonia.scm.security.AllowAnonymousAccess;
 
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
 import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
 import javax.ws.rs.Path;
-import javax.ws.rs.core.Response;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import java.util.List;
+
+import static de.otto.edison.hal.Link.link;
+import static de.otto.edison.hal.Links.linkingTo;
+import static java.util.stream.Collectors.toList;
 
 @AllowAnonymousAccess
 @Path("v2/smeagol/")
 public class SmeagolResource {
 
+  private final SmeagolRepositoryStore store;
+  private final SmeagolLinkBuilder smeagolLinkBuilder;
+  private final SmeagolConfiguration configuration;
+  private final SmeagolConfigurationDtoMapper configurationMapper;
+
+  @Inject
+  SmeagolResource(SmeagolRepositoryStore store, SmeagolLinkBuilder smeagolLinkBuilder, SmeagolConfiguration configuration, SmeagolConfigurationDtoMapper configurationMapper) {
+    this.store = store;
+    this.smeagolLinkBuilder = smeagolLinkBuilder;
+    this.configuration = configuration;
+    this.configurationMapper = configurationMapper;
+  }
+
   @GET
-  @Path("resources")
-  public Response loadRepositories() {
-    return Response.ok("to be done").build();
+  @Path("repositories")
+  @Produces("application/json")
+  public HalRepresentation loadRepositories(@QueryParam("wikiEnabled") boolean smeagolOnly) {
+    List<SmeagolRepositoryInformationDto> repositories = store.getRepositories();
+    if (smeagolOnly) {
+      repositories = repositories.stream().filter(SmeagolRepositoryInformationDto::isWikiEnabled).collect(toList());
+    }
+    return new HalRepresentation(
+      linkingTo().single(link("self", smeagolLinkBuilder.getRepositoriesLink())).build(),
+      Embedded.embedded("repositories", repositories)
+    );
```

Comment:
I think there a missing links on the `SmeagolRepositoryInformationDto`. The dto should have at least on link for the ui of the repository, to allow smeagol to create a back link for scm-manager. A nice addition would be a link which points to rest api endpoint for the repository.

Reply:
Added the link to the UI. Will wait for detailed requirements until adding further links, because this may require core changes.